### PR TITLE
Component: Add Pricing Slider component

### LINKED ISSUE #31345: Paid Stats / Components: Pricing Slider
Please refer to the implementation on Akismet to accelerate the development: https://akismet.com/plan/personal/

Typical props would include: `defaultValue`, `value`, `minStep`, `minValue`, `maxValue`, `onChange`, `sliderColor` etc.

Image

### PR DESCRIPTION
**Pull request by dognose24**
Fixes #31345 

## Proposed changes:

* Add component `Pricing Slider` with third-party library [ReactSlider](https://zillow.github.io/react-slider/#reactslider).
* Add storybook use cases for operations.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1HpG7-nar-p2

## Does this pull request change what data or activity we track or use?

N/A.

## Testing instructions:

* Run the local storybook with `cd projects/js-packages/storybook && npm run storybook:dev`.
* Ensure the `Pricing Slider` component under `JS Packages` works well.

截圖 2023-06-19 下午12 04 03

### CODE REVIEW

File: projects/js-packages/components/components/pricing-slider/index.tsx
Comment (kangzj) [inline projects/js-packages/components/components/pricing-slider/index.tsx:5]: Can we please change it to use CSS module: https://github.com/Automattic/jetpack/blob/49b12cd171952f0ab65c5a2d92901b65644de073/projects/js-packages/components/components/action-button/index.jsx#L13

**Comment by kangzj** [comment]
I fixed the build with c90c302 and I think we should use CSS module just like other components to avoid CSS conflicts. Commented inline.

It otherwise works well for me! You are a legend @dognose24 👍

File: projects/js-packages/components/components/pricing-slider/style.scss
Comment (jsnmoon) [inline projects/js-packages/components/components/pricing-slider/style.scss]: I think we need to fix the indentation style in this file, which uses both spaces and tabs.

File: projects/js-packages/components/components/pricing-slider/style.scss
Comment (jsnmoon) [inline projects/js-packages/components/components/pricing-slider/style.scss]: Missing an empty line at the end.
```suggestion
}

```

File: projects/js-packages/components/components/pricing-slider/types.ts
Comment (jsnmoon) [inline projects/js-packages/components/components/pricing-slider/types.ts]: This needs to be updated to resolve a typing error.

Comment (jsnmoon) [reply projects/js-packages/components/components/pricing-slider/types.ts]: Pushed up an attempted fix with [332a5ea](https://github.com/Automattic/jetpack/pull/31425/commits/332a5ea8b2074d57ef0e7739c86ee49d2afbda09).

File: projects/js-packages/components/components/pricing-slider/style.scss
Comment (jsnmoon) [reply projects/js-packages/components/components/pricing-slider/style.scss]: Fixed in [2c43d33](https://github.com/Automattic/jetpack/pull/31425/commits/2c43d3362265d02461b4129abf4f87f093c47df9).

File: projects/js-packages/components/components/pricing-slider/style.scss
Comment (jsnmoon) [reply projects/js-packages/components/components/pricing-slider/style.scss]: Fixed in [2c43d33](https://github.com/Automattic/jetpack/pull/31425/commits/2c43d3362265d02461b4129abf4f87f093c47df9).

**Review by jsnmoon — DISMISSED** [review]
Looks great! I pushed up two commits to resolve my review nits and the build failure. 🚢

File: projects/js-packages/components/components/pricing-slider/index.tsx
Comment (kangzj) [reply projects/js-packages/components/components/pricing-slider/index.tsx:5]: `react-slider` appends `-0` and `-1` to for slider colours, so it's hard to leverage CSS modules. We agreed to change the class names with good / long BEM class names to avoid any conflicts.

**Comment by dognose24** [comment]
> use CSS module just like other components to avoid CSS conflicts

@kangzj I agree that the CSS conflict could affect more than we imagine. However, CSS modules would not be adaptable to the library since the mechanism of dynamically determining class names for tracks (`${trackClassName}-${i}`).

I updated class names with more detailed BEM styles to avoid CSS conflict. [9cd3acf](https://github.com/Automattic/jetpack/pull/31425/commits/9cd3acfad066505968b7dce772a213e63aabc7b9)

cc @jsnmoon

**Review by kangzj — DISMISSED** [review]
It works well for me. Great job! 👍

Let's try to fix the build errors and then we could ship 🐑

File: projects/packages/backup/webpack.config.js
Comment (anomiex) [inline projects/packages/backup/webpack.config.js]: This should not be necessary. I'm guessing this is trying to work around the missing peer dep but with an inaccurate comment?

Please remove this (here and everywhere else you added it) when trying the fix I suggest in the top-level comment.

File: pnpm-lock.yaml
Comment (anomiex) [inline pnpm-lock.yaml:1]: You've apparently done something weird with the dependency updates here, there are way too many package versions being bumped here.

I recommend starting over here by making sure any dependencies you're adding on existing packages match those used already in use in the monorepo, do a trunk merge, then `git checkout origin/trunk pnpm-lock.yaml && pnpm install` to update this afresh.

Although you may want to wait until I finish getting #31308 merged before doing that, which I'm working on right now.

**Review by anomiex — DISMISSED** [review]
It looks like the react-slider library you're trying to use has an undeclared peer dependency on `@babel/runtime`, similar to [this bug in Gutenberg](https://href.li/?https://github.com/WordPress/gutenberg/issues/41343). An upstream bug should be filed for that, and then you could do something like [this](https://github.com/Automattic/jetpack/blob/95b8fab93bde8973940e5fc397582673d8a9ef6e/.pnpmfile.cjs#L46-L54) (including a comment linking to the upstream bug) to work around the bug if you don't want to wait for upstream to release a fix.

**Comment by dognose24** [comment]
@anomiex Thanks for your direction a lot! 🙇🏼‍♂️

> An upstream bug should be filed for that

There was an [upstream issue reported](https://github.com/zillow/react-slider/issues/284), but it seemed to be ignored by the author.

> then you could do something like [this](https://github.com/Automattic/jetpack/blob/95b8fab93bde8973940e5fc397582673d8a9ef6e/.pnpmfile.cjs#L46-L54) (including a comment linking to the upstream bug) to work around the bug

So, could we apply the workaround for now? 🙏🏼

**Comment by anomiex** [comment]
> There was an [upstream issue reported](https://github.com/zillow/react-slider/issues/284), but it seemed to be ignored by the author.

It doesn't look ignored to me so much as that the person there found that, in a hoisted environment, adding the missing peer dep worked around it so they closed it.

A bug report explicitly pointing out that it's a missing peer dependency and that the workaround of just adding the dependency won't work with pnpm with hoisting disabled or with yarn's PnP would be good to try. Feel free to crib from the Gutenberg bug I linked.

> So, could we apply the workaround for now? 🙏🏼

Yes, but please do file the upstream bug for the comment to link to.

File: projects/packages/backup/webpack.config.js
Comment (jsnmoon) [reply projects/packages/backup/webpack.config.js]: These changes have been reverted.

File: pnpm-lock.yaml
Comment (jsnmoon) [reply pnpm-lock.yaml:1]: This has been resolved.

**Comment by jsnmoon** [comment]
This pull request should now be ready for another round of reviews.

**Review by kangzj — APPROVED** [review]
LGTM 👍

**Pull request merged**
